Feat implement proposal cost and min rep

BenjaminMeijs opened the pull request:
[Summary of the change]

# Changes

- [x] Added useFacetFetch hook
- [x] replaced useVotingSettings with a useFacetFetch implementation
- [x] replaced useTimeClaimable with a useFacetFetch implementation
- [x] added propocal cost and minimum rep to the confirmation page of new proposal form

ThijnK commented on src/components/newProposal/steps/Confirmation.tsx:
```diff
@@ -274,8 +298,50 @@ export const Confirmation = () => {
+          proposalCreationCost !== null &&
+          votingPower.gte(proposalCreationCost) && (
+            <p>
+              Creating a proposal costs{' '}
```
I would personally prefer "You will pay X SECOREP"
But I leave it up to you if you actually choose to replace this with it

ThijnK commented on src/components/newProposal/steps/Voting.tsx:
```diff
@@ -70,7 +67,8 @@ export type EndTimeType = 'duration' | 'end-custom';
 
-  const { settings, error } = useVotingSettings();
+  const { data: minDurationBN, error } = usePartialVotingProposalMinDuration();
+  const minDuration = minDurationBN?.toNumber() ?? null;
```
Same as I mentioned above. Maybe we ought to actually catch possible errors with the `toNumber()` conversion

ThijnK reviewed (changes requested):
There's some very minor stuff

ThijnK reviewed (approved):
Applied my own feedback already
Love the `useFacetFetch` hook, good stuff

ThijnK reviewed (approved):
Already approved above

BenjaminMeijs commented:
@ThijnK  I've added some extra stuff:
- added useTier hook
- added useTieredTimeClaimable hook
- changed useFacetFetch to allow for context (extra powerfull stuff)
- added a simple progress bar to the ClaimDailyReward card. Due to SDK issues I could not yet test this. Perhaps remove in the future

ThijnK commented on src/components/dashboard/ClaimDailyRewardCard.tsx:
Looking at the progress bar in the UI, I feel like it might be confusing to users. What is it? What do it meaaan?
I  imagine many people would assume something like it being a progress bar counting down to the next claim moment (which in a way is true), but actually (afaik) it's a buildup to the max that you can have stashed for claiming.

What I'm getting at is that we should probably add some text or something that indicated that it is a counter of how many days worth of daily rewards you have available for claiming

ThijnK commented on src/hooks/useTier.ts:
```diff
@@ -0,0 +1,50 @@
+import { useDiamondSDKContext } from '../context/DiamondGovernanceSDK';
+import { getErrorMessage } from '../lib/utils';
+
+export const useTier = () => {
```
Can we have a comment here to explain what this hook is?

The other hooks in useFacetFetch.tsx I feel speak for themselves mostly, but this one's a bit less trivial (at least to me)

ThijnK reviewed (changes requested):
See comment below
